### PR TITLE
Part E of transition to support modernize-use-default-member-init

### PR DESCRIPTION
**Pull request by gnawme**
Part E of #5665 to add these checks to clang-tidy:

-     [modernize-use-default-member-init](https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-default-member-init.html)
-     [modernize-use-bool-literals](https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-bool-literals.html)

### CODE REVIEW

**Comment by mvieth** [comment]
Could you resolve the conflict, please?

File: features/include/pcl/features/brisk_2d.h [hunk 1]
```diff
@@ -176,7 +176,7 @@ namespace pcl
       KeypointPointCloudTPtr keypoints_;
 
       // TODO: set
-      float scale_range_;
+      float scale_range_{};
```
Comment (larshg) [inline features/include/pcl/features/brisk_2d.h:179]: Could we add default values here as well? Even if it is 0 / 0.0 etc.

File: keypoints/include/pcl/keypoints/smoothed_surfaces_keypoint.h [hunk 2]
```diff
@@ -116,14 +113,14 @@ namespace pcl
...
+      float neighborhood_constant_{0.5f};
       std::vector<PointCloudTConstPtr> clouds_;
       std::vector<PointCloudNTConstPtr> cloud_normals_;
       std::vector<KdTreePtr> cloud_trees_;
       PointCloudNTConstPtr normals_;
       std::vector<std::pair<float, std::size_t> > scales_;
-      float input_scale_;
-      std::size_t input_index_;
+      float input_scale_{0.0f};
+      std::size_t input_index_{};
```
Comment (larshg) [inline keypoints/include/pcl/keypoints/smoothed_surfaces_keypoint.h:123]:
```suggestion
      std::size_t input_index_{0};
```

File: ml/include/pcl/ml/dt/decision_tree_trainer.h [hunk 3]
```diff
@@ -229,11 +229,11 @@ class PCL_EXPORTS DecisionTreeTrainer {
...
+  std::size_t max_tree_depth_{15};
   /** Number of features used to find optimal decision features. */
-  std::size_t num_of_features_;
+  std::size_t num_of_features_{1000};
   /** Number of thresholds. */
-  std::size_t num_of_thresholds_;
+  std::size_t num_of_thresholds_{10};
 
   /** FeatureHandler instance, responsible for creating and evaluating features. */
   pcl::FeatureHandler<FeatureType, DataSet, ExampleIndex>* feature_handler_;
```
Comment (larshg) [inline ml/include/pcl/ml/dt/decision_tree_trainer.h:239]:
```suggestion
  pcl::FeatureHandler<FeatureType, DataSet, ExampleIndex>* feature_handler_{nullptr}; 
```
And removed from the constructor initializer list?

Comment (larshg) [reply ml/include/pcl/ml/dt/decision_tree_trainer.h:239]: There are more fields that can be done the same with, ie. stats_estimator_?

I'm not sure about the vector types, if they can be with just {} to create the empty vectors? ie. label_data_, examples_ and the data_set_ (which is of DataSet type)

File: outofcore/include/pcl/outofcore/outofcore_node_data.h [hunk 4]
```diff
@@ -176,7 +176,7 @@ namespace pcl
         /** \brief Metadata (JSON) file pathname (oct_idx extension JSON file) */
         boost::filesystem::path metadata_filename_;
         /** \brief Outofcore library version identifier */
-        int outofcore_version_;
+        int outofcore_version_{};
```
Comment (larshg) [inline outofcore/include/pcl/outofcore/outofcore_node_data.h:179]:
```suggestion
        int outofcore_version_{0};
```

File: features/include/pcl/features/brisk_2d.h [hunk 5]
```diff
@@ -214,41 +214,41 @@ namespace pcl
       BriskPatternPoint* pattern_points_;
       
       /** Total number of collocation points. */
-      unsigned int points_;
+      unsigned int points_{};
```
Comment (larshg) [inline features/include/pcl/features/brisk_2d.h:217]:
```suggestion
      unsigned int points_{0};
```

File: outofcore/include/pcl/outofcore/outofcore_node_data.h [hunk 6]
```diff
@@ -176,7 +176,7 @@ namespace pcl
         /** \brief Metadata (JSON) file pathname (oct_idx extension JSON file) */
         boost::filesystem::path metadata_filename_;
         /** \brief Outofcore library version identifier */
-        int outofcore_version_;
+        int outofcore_version_{0u};
```
Comment (larshg) [inline outofcore/include/pcl/outofcore/outofcore_node_data.h:179]:
```suggestion
        int outofcore_version_{0};
```
?

File: ml/include/pcl/ml/impl/dt/decision_tree_trainer.hpp [hunk 7]
```diff
@@ -46,16 +46,7 @@ template <class FeatureType,
...
-, num_of_features_(1000)
-, num_of_thresholds_(10)
-, feature_handler_(nullptr)
-, stats_estimator_(nullptr)
-, data_set_()
-, label_data_()
-, examples_()
-, decision_tree_trainer_data_provider_()
-, random_features_at_split_node_(false)
+: data_set_(), decision_tree_trainer_data_provider_()
```
Comment (larshg) [inline ml/include/pcl/ml/impl/dt/decision_tree_trainer.hpp:49]:
```suggestion
: data_set_()
```

File: ml/include/pcl/ml/dt/decision_tree_trainer.h [hunk 8]
```diff
@@ -229,39 +229,40 @@ class PCL_EXPORTS DecisionTreeTrainer {
...
   /** FeatureHandler instance, responsible for creating and evaluating features. */
-  pcl::FeatureHandler<FeatureType, DataSet, ExampleIndex>* feature_handler_;
+  pcl::FeatureHandler<FeatureType, DataSet, ExampleIndex>* feature_handler_{nullptr};
   /** StatsEstimator instance, responsible for gathering stats about a node. */
-  pcl::StatsEstimator<LabelType, NodeType, DataSet, ExampleIndex>* stats_estimator_;
+  pcl::StatsEstimator<LabelType, NodeType, DataSet, ExampleIndex>* stats_estimator_{
+      nullptr};
 
   /** The training data set. */
   DataSet data_set_;
```
Comment (larshg) [inline ml/include/pcl/ml/dt/decision_tree_trainer.h:245]:
```suggestion
  DataSet data_set_{};
```

File: ml/include/pcl/ml/impl/dt/decision_tree_trainer.hpp (hunk 7 above)
Comment (larshg) [reply ml/include/pcl/ml/impl/dt/decision_tree_trainer.hpp:49]: Depends on DataSet data_set_{}; being set aswell.

**Review by larshg — APPROVED** [review]

**Comment by larshg** [comment]
Looks good, thanks :+1:

**Review by mvieth — APPROVED** [review]
Thanks!

**Pull request merged**